lib/node-package-manager.ts

```diff
@@ -8,139 +7,99 @@ interface INpmOpts {
-				config["ignore-scripts"] = true;
+		try {
+			let subDirsBefore:Array<string> = this.$fs.exists(path.join(pathToSave, "node_modules")).wait() ? this.$fs.readDirectory(path.join(pathToSave, "node_modules")).wait() : [];
+
+			let flags = this.getFlagsString(config);
+			this.$childProcess.exec(`npm install ${packageName} ${flags}`, { cwd: pathToSave }).wait();
+
+			let subDirsAfter = this.$fs.readDirectory(path.join(pathToSave, "node_modules")).wait();
+
+			/** This diff is done in case the installed pakcage is a URL address, a path to local directory or a .tgz file
+			 *  in these cases we don't have the package name and we can't rely on "npm install --json"" option
+			 *  to get the project name because we have to parse the output from the stdout and we have no controll over it (so other messages may be mangled in stdout)
+			 * 	The solution is to traverse node_modules before and after install and get the name of the installed package,
+			 * 	even if it's installed through local path or URL. If command installes more than one package, all package names are returned.
+			 */
+			let diff = subDirsAfter.filter(x => subDirsBefore.indexOf(x) === -1);
+			if(diff.length <= 0 && subDirsBefore.length !== subDirsAfter.length) {
+				this.$errors.failWithoutHelp(`Couldn't install package correctly`);
 			}
 
-			try {
-				return this.loadAndExecute("install", [pathToSave, packageName], { config: config }).wait();
-			} catch (err) {
-				if (err.code === "EPEERINVALID") {
-					// Not installed peer dependencies are treated by npm 2 as errors, but npm 3 treats them as warnings.
-					// We'll show them as warnings and let the user install them in case they are needed.
-					// The strucutre of the error object in such case is:
-					//	{ [Error: The package @angular/core@2.1.0-beta.0 does not satisfy its siblings' peerDependencies requirements!]
-					//   code: 'EPEERINVALID',
-					//   packageName: '@angular/core',
-					//   packageVersion: '2.1.0-beta.0',
-					//   peersDepending:
-					//    { '@angular/common@2.1.0-beta.0': '2.1.0-beta.0',
-					//      '@angular/compiler@2.1.0-beta.0': '2.1.0-beta.0',
-					//      '@angular/forms@2.1.0-beta.0': '2.1.0-beta.0',
-					//      '@angular/http@2.1.0-beta.0': '2.1.0-beta.0',
-					//      '@angular/platform-browser@2.1.0-beta.0': '2.1.0-beta.0',
-					//      '@angular/platform-browser-dynamic@2.1.0-beta.0': '2.1.0-beta.0',
-					//      '@angular/platform-server@2.1.0-beta.0': '2.1.0-beta.0',
-					//      '@angular/router@3.1.0-beta.0': '2.1.0-beta.0',
-					//      '@ngrx/effects@2.0.0': '^2.0.0',
-					//      '@ngrx/store@2.2.1': '^2.0.0',
-					//      'ng2-translate@2.5.0': '~2.0.0' } }
-					this.$logger.warn(err.message);
-					this.$logger.trace("Required peerDependencies are: ", err.peersDepending);
-				} else {
-					// All other errors should be handled by the caller code.
-					throw err;
-				}
+			return diff;
+		} catch (err) {
+			if (err.code === "EPEERINVALID") {
```

Comment:
Since we're not using npm as a library anymore, will we actually get an exception with an EPEERINVALID `code` from the child process? Seems like we should either delete this code or handle invalid peer dependencies in another way.